install/ui/src/freeipa/widgets/MigrateScreen.js

```diff
@@ -0,0 +1,215 @@
+                var u_f = this.get_field('username');
+                var p_f = this.get_field('password');
+                var required = !util.is_empty(u_f.get_value()) ||
+                    !util.is_empty(p_f.get_value());
+                u_f.set_required(required);
+                p_f.set_required(required);
+            },
+
+            on_confirm: function() {
+                this.migrate();
+            },
+
+            migrate: function() {
+                var val_summary = this.get_widget('validation');
+                val_summary.remove('migrate');
+
+                if (!this.validate()) return;
+
+                var username = this.get_field('username');
+                var psw = this.get_field('password');
+                var result = this.migrate_core(
+                    username.get_value()[0],
+                    psw.get_value()[0]);
+
+                psw.set_value('');
+                if (result.status === 'ok') {
+                    val_summary.add_success('migrate', this.migration_success);
+                    window.setTimeout(this.redirect, 3000);
+                } else {
+                    val_summary.add_error('migrate', result.message);
+                }
+            },
+
+            redirect: function() {
+                window.location = config.url;
+            },
+
+            migrate_core: function(username, password) {
+
+                //possible results: 'ok', 'invalid-password', 'migration-error'
+
+                var status = 'invalid';
+                var result = {
+                    status: status,
+                    message: this.migration_failure_msg
+                };
+
+                function success_handler(data, text_status, xhr) {
+                    topic.publish('rpc-end');
+                    result.status = xhr.getResponseHeader(
+                        "X-IPA-Migrate-Result") || status;
+
+                    if (result.status === 'migration-error') {
+                        result.message = this.migration_error_msg;
+                    } else if (result.status === 'invalid-password') {
+                        result.message = this.migration_invalid_password;
+                    }
+                    return result;
+                }
+
+                function error_handler(xhr, text_status, error_thrown) {
+                    topic.publish('rpc-end');
+                    return result;
+                }
+
+                var data = {
+                    username: username,
+                    password: password
+                };
+
+                var request = {
+                    url: config.migration_url,
+                    data: data,
+                    contentType: 'application/x-www-form-urlencoded',
+                    processData: true,
+                    dataType: 'html',
+                    async: false,
+                    type: 'POST',
+                    success: success_handler,
```

Comment:
the success and error handlers are not bound to the MigrateScreen widget object. Thus, in their execution, `this` object will be the http request and so no message will be displayed at error.